### PR TITLE
Implemented Update Task API

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
### Description of Change
- Added Put request handler at controller
- Implemented update task functionality at service
### Associated Ticket Link:
- https://trello.com/c/mRyqzqkg

### CODE REVIEW

File: src/tasks/tasks.service.ts [hunk 1]
```diff
@@ -21,4 +21,13 @@ export class TasksService {
     await this.em.persistAndFlush(task);
     return task;
   }
+
+  async updateTask(id: number, updateTaskDto: UpdateTaskDto): Promise<Task> {
+    const task = await this.taskRepository.findOne(id);
+    if (task) {
+      this.taskRepository.assign(task, updateTaskDto);
+      await this.em.persistAndFlush(task);
```
Comment (ishmam-iqbal-sazim) [inline src/tasks/tasks.service.ts:29]: Let's do this in a method in the repository layer and call that here.

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comment

File: src/tasks/tasks.service.ts (hunk 1 above)
Comment (faysal-alam-sazim) [reply src/tasks/tasks.service.ts:30]: Bhai what about other methods like get all tasks, create a task? Should I leave these in services?

File: src/tasks/tasks.repository.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import { Task } from 'src/common/entities/task.entity';
+import { UpdateTaskDto } from './tasks.dto';
+
+@Injectable()
+export class TaskRepository extends EntityRepository<Task> {
+  constructor(em: EntityManager) {
+    super(em, Task);
+  }
+
+  async updateTask(id: number, updateTaskDto: UpdateTaskDto): Promise<Task> {
```
Comment (ragib-hossain-sazim) [inline src/tasks/tasks.repository.ts:12]: [Nitpick] `updateTask` => `updateOne`

File: src/tasks/tasks.repository.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    const task = await this.findOne(id);
```
Comment (ragib-hossain-sazim) [inline src/tasks/tasks.repository.ts:13]: `findOne` => `findOneOrFail` also this is business logic. Move this to service layer. This method should only have code related to updating.

File: src/tasks/tasks.repository.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    if (task) {
```
Comment (ragib-hossain-sazim) [inline src/tasks/tasks.repository.ts:14]: this condition won't be required if you use `findOrFail` inside `tasks.service.ts`

File: src/tasks/tasks.service.ts [hunk 5]
```diff
@@ -1,15 +1,14 @@
...
+import { EntityManager, EntityRepository } from '@mikro-orm/postgresql';
+import { InjectRepository } from '@mikro-orm/nestjs';
+import { TaskRepository } from './tasks.repository';
 
 @Injectable()
 export class TasksService {
   constructor(
-    @InjectRepository(Task)
-    private readonly taskRepository: EntityRepository<Task>,
+    private readonly taskRepository: TaskRepository,
```
Comment (ragib-hossain-sazim) [inline src/tasks/tasks.service.ts:11]: `taskRepository` => `tasksRepository`

File: src/tasks/tasks.repository.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { EntityManager, EntityRepository } from '@mikro-orm/postgresql';
+import { Injectable } from '@nestjs/common';
+import { Task } from 'src/common/entities/task.entity';
+import { UpdateTaskDto } from './tasks.dto';
+
+@Injectable()
+export class TaskRepository extends EntityRepository<Task> {
```
Comment (ragib-hossain-sazim) [inline src/tasks/tasks.repository.ts:7]: `TaskRepository` => `TasksRepository`

**Review by ragib-hossain-sazim — APPROVED** [review]

**Pull request merged**